class.jetpack-gutenberg.php

```diff
@@ -87,6 +112,20 @@ private static function prepend_block_prefix( $block_name ) {
 		return 'jetpack/' . $block_name;
 	}
 
+	/**
+	 * Remove the 'jetpack-' prefix from an extension name
+	 *
+	 * @param string $extension_name The extension name.
+	 *
+	 * @return string The prefixed block name.
+	 */
+	private static function remove_extension_prefix( $extension_name ) {
+		if ( wp_startswith( $extension_name, 'jetpack/' ) ) {
```

Comment:
can we add a filter here so that this warns developers in development mode?

```php
private static function remove_extension_prefix( $extension_name ) {
	if ( ! wp_startswith( $extension_name, 'jetpack/' ) ) {
		return $extension_name;
	}

	do_action( 'jetpack_gutenberg_extension_missing_prefix', $extension_name );
	return substr( $extension_name, strlen( 'jetpack/' ) );
}
```

maybe my logic is bad or I don't understand what's going on. my point is that maybe we can do something where in our local machines during development work we can hook into this action and either kill the script or flag a big warning somehow to help prevent us from propagating two ways of naming things.

Reply:
I'm hoping to hard-wire the prefixes back in soon after landing this PR. TBH I don't know what kind of development mode tooling JP offers (does it?) so I'd rather forgo for now :grimacing: I promise to keep an eye on these.